sfepy/discrete/fem/refine.py

```diff
@@ -7,6 +7,33 @@
 from sfepy.discrete.fem import Mesh
 from six.moves import range
 
+
+def refine_1_2(mesh_in):
+    """
+    Refines 1D mesh by cutting each element in half
+    """
+
+    if not nm.all(mesh_in.coors[:-1] <= mesh_in.coors[1:]):
+        raise ValueError("1D Mesh for refinement must have sorted coors array")
+
+    cmesh = mesh_in.cmesh
+    c_centres = cmesh.get_centroids(cmesh.dim)
+    new_coors = nm.zeros((2*mesh_in.n_nod - 1, 1))
+    new_coors[0::2] = mesh_in.coors
+    new_coors[1::2] = c_centres
+
+    n_nod = len(new_coors)
+
+    new_conn = nm.arange(n_nod, dtype=nm.int32).repeat(2)[1:-1].reshape((-1, 2))
+
+    if not nm.all(cmesh.cell_groups[0] == cmesh.cell_groups):
+        raise Warning("Multiple mat ids not supported when refining, using cmesh.cell_groups[0] == {}".format(cmesh.cell_groups[0]))
+    new_mat_id = nm.zeros(n_nod - 1, dtype=nm.int32) + cmesh.cell_groups[0]
```

Comment:
It should be possible to preserve the cell groups like in the other refine functions.

Reply:
Ok it was actually quite simple.